lib/airtap.js

```diff
@@ -30,39 +30,20 @@ function Zuul (config) {
     config.browser_open_timeout = 120 * 1000
   }
 
-  var self = this
-
-  var ui = config.ui
-  var framework_dir = frameworks[ui]
-  if (!framework_dir) {
-    throw new Error('unsupported ui: ' + ui)
-  }
-
-  config.framework_dir = framework_dir
-  self._config = config
+  // TODO hardcoding this to 'tape' for now, but shouldn't be configurable at
+  // all, which this means we should let control-app.js take care of this
+  config.framework_dir = frameworks.tape
+  this._config = config
 
   debug('config: %j', omit(config, ['sauce_username', 'sauce_key', 'username', 'key']))
 
   // list of browsers to test
-  self._browsers = []
-
-  self._concurrency = config.concurrency || 5
+  this._browsers = []
+  this._concurrency = config.concurrency || 5
 }
 
 Zuul.prototype.__proto__ = EventEmitter.prototype
 
-Zuul.prototype._setup = function (cb) {
-  var self = this
-
-  var config = self._config
-
-  // we only need one control app
-  var control_server = control_app(config).listen(0, function () {
-    debug('control server active on port %d', control_server.address().port)
-    cb(null, control_server.address().port)
-  })
-}
-
```

Comment:
This method adds no value. Just call `createControlServer` from `.run()` directly